src/lib/mpq.mjs

```diff
@@ -91,9 +100,15 @@ class MPQ {
     }
   }
 
-  static async open(path, flags = 0) {
+  static async open(path, mode = '') {
     await StormLib.ready;
 
+    let flags = 0;
+
+    if (mode === 'r') {
+      flags |= StormLib.STREAM_FLAG_READ_ONLY;
+    }
```

Comment:
Beautiful 👍

Reply:
I'm intending to add some jsdoc annotations soon, too, so these values aren't a mystery.